contrib/opencensus-ext-azure/examples/logs/contextual_logging.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2019, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from opencensus.ext.azure.log_exporter import AzureLogHandler
+
+logger = logging.getLogger(__name__)
+# TODO: you need to specify the instrumentation key in a connection string
+# and place it in the APPLICATIONINSIGHTS_CONNECTION_STRING
+# environment variable.
+logger.addHandler(AzureLogHandler())
+
+# For more details on contextual logging patterns in python, see:
+# https://docs.python.org/3/howto/logging-cookbook.html#adding-contextual-information-to-your-logging-output
+
+'''
+Option 1: Use LoggerAdapters to add custom dimensions
+'''
+
+
+class CustomDimensionsAdapter(logging.LoggerAdapter):
+
+    @property
+    def custom_dimensions(self):
+        if self.extra and 'custom_dimensions' in self.extra:
+            return self.extra['custom_dimensions']
+        else:
+            return {}
+
+    def process(self, msg, kwargs):
+
+        if self.custom_dimensions:
+            if 'extra' not in kwargs:
+                kwargs['extra'] = {'custom_dimensions': self.custom_dimensions}
+
+            elif 'extra' in kwargs and 'custom_dimensions' in kwargs['extra']:
+                kwargs['extra'] = {
```

Comment:
Would this overwrite other entries in the `extra` dict?

Reply:
Good catch! You are right!

Reply:
Another questions is that this is currently a sample of how someone can add contextual logging easily to their code, however, they would still need to implement these classes and support them (even though they are just related to updating a properties field for log export. Is it worthwhile to consider supporting these two classes as part of the package?

Reply:
The `Azure` package supports solely on features that relate to sending telemetry to Azure Monitor. How the user wants to do this is up to them. We might make the `properties` feature more robust in the future (by adding the classes as you've mentioned) but I think for now it is fine to just include them in the example file.

Reply:
Makes sense!